rust/src/dataset.rs

```diff
@@ -1119,6 +1123,41 @@ mod tests {
         assert_eq!(&ArrowSchema::from(first_ver.schema()), schema.as_ref());
     }
 
+    #[tokio::test]
+    async fn test_commit_append() {
+        let test_dir = tempdir().unwrap();
+
+        let schema = Arc::new(ArrowSchema::new(vec![Field::new(
+            "i",
+            DataType::Int32,
+            false,
+        )]));
+        let batches = RecordBatchBuffer::new(vec![RecordBatch::try_new(
+            schema.clone(),
+            vec![Arc::new(Int32Array::from_iter_values(0..20))],
+        )
+        .unwrap()]);
+
+        let test_uri = test_dir.path().to_str().unwrap();
+        let mut write_params = WriteParams::default();
+        write_params.max_rows_per_file = 40;
+        write_params.max_rows_per_group = 10;
+        let mut batches: Box<dyn RecordBatchReader> = Box::new(batches);
+        let dataset = Dataset::write(&mut batches, test_uri, Some(write_params))
+            .await
+            .unwrap();
+        assert_eq!(20, dataset.count_rows().await.unwrap());
+
+        let fragments: Vec<Fragment> = dataset.fragments().iter().map(|f| f.clone()).collect();
+
+        let new_dataset =
+            Dataset::commit(test_uri, dataset.schema(), &fragments, WriteMode::Append)
+                .await
+                .unwrap();
```

Comment:
I don't think this should be allowed because the added fragments have the same ids as the old ones.